linux/hid.c

```diff
@@ -566,6 +734,52 @@ struct hid_device_info  HID_API_EXPORT *hid_enumerate(unsigned short vendor_id,
 					 * check for USB and Bluetooth devices above */
 					break;
 			}
+
+			/* Usage Page and Usage */
+			int res;
+			struct hidraw_report_descriptor rpt_desc;
+			int device_handle = open(dev_path, O_RDWR);
+			if (device_handle > 0) {
```

Comment:
```suggestion
			if (device_handle >= 0) {
```
`0` - is a valid file descriptor and may be returned by `open`

Reply:
Pushed the change (and rebased on top of master)